src/Game/UI/Gumps/HealthBarGumpCustom.cs

```diff
@@ -0,0 +1,662 @@
+// be it (straight line(custom) or graphic(classic) is not directly included in this version
+// however it will use your default settings(art modifications) if needed.
+//
+// I purposefully did not included my Notoriety Organizer to this version as I am still waiting
+// to hear back from shard owners &/OR KaRaShO to respond regarding the potential abuse of this.
+// Also I made a poll of 100(exactly) people whether or not they wanted healthbar buttons to be
+// added into the custom bars when in party mode, the results were 93-7 in favor of not having
+// them included. On my current version I do have this added if more desire for this ever comes.
+//
+// There should be no problems with this version as I've thoroughly tested it but if anything does 
+// crash or seems like a bug/issue, please DM me on Discord at Alan#0084 and I will get it fixed.
+// At present the only known issue is when using the Custom Health Bars and renaming a pet. The text
+// is rendered so the option to edit that has been added as a bandaid but not fully integrated. You
+// can rename pets, the entry cursor will reside to the left side of the bar but it will not remove
+// the current name. You will need to type enter and re-pull the bar to see the name change.
+//
+// Lastly, I want to give thanks to KaRaShO, Roxya, Stalli, Gaechti, and Link for giving me tips on how
+// to get these Health Bars to function per my own vision; gratitude.
+
+#endregion
+
+using System;
+using System.IO;
+
+using ClassicUO.Game.Data;
+using ClassicUO.Game.GameObjects;
+using ClassicUO.Game.Managers;
+using ClassicUO.Game.UI.Controls;
+using ClassicUO.Input;
+using ClassicUO.IO;
+using ClassicUO.IO.Resources;
+using ClassicUO.Network;
+using ClassicUO.Renderer;
+
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using SDL2;
+
+namespace ClassicUO.Game.UI.Gumps
+{
+    internal class HealthBarGumpCustom : AnchorableGump
+    {
+
+        private const ushort LINE_RED = 0x0805;
+        private const ushort LINE_BLUE = 0x0806;
+        private const ushort LINE_POISONED = 0x0808;
+        private const ushort LINE_YELLOWHITS = 0x0809;
+        private const int MIN_WIDTH = 120;
+
+        private readonly GumpPicWithWidth[] _bars = new GumpPicWithWidth[3];
+        private GumpPic _hpLineRed, _manaLineRed, _stamLineRed;
+
+        private bool _canChangeName;
+        private bool _isDead;
+        private bool _oldWarMode, _normalHits, _poisoned, _yellowHits;
+        private bool _outOfRange;
+        private bool _targetBroke;
+
+        private string _name;
+        private int _oldHits, _oldStam, _oldMana;
+        private TextBox _textBox;
+
+        public RenderedText _renderedText;
+        public AlphaBlendControl _background;
+
+        public HealthBarGumpCustom(Entity entity) : this()
+        {
+            if (Engine.Profile.Current.CustomBarsToggled == false)
+            {
+                Engine.UI.GetGump<HealthBarGump>(entity.Serial);
```

Comment:
What's happening here?
You get the object and...? Missing a dispose maybe?

Reply:
It's to double-check to make sure it's not healthbargump, the dispose is a line down.

Reply:
But that code does litterally nothing, i think you can remove it